protocol/v2/blockchain/beacon/network.go

```diff
@@ -42,28 +40,82 @@ type BeaconNetwork interface {
 	GetBeaconNetwork() spectypes.BeaconNetwork
 }
 
-// NewNetwork creates a new beacon chain network.
+// Network is a beacon chain network.
+type Network struct {
+	Parent                          spectypes.BeaconNetwork `json:"parent"`
+	Name                            string                  `json:"name"`
+	ForkVersionVal                  [4]byte                 `json:"fork_version"`
+	MinGenesisTimeVal               uint64                  `json:"min_genesis_time"`
+	SlotDurationSecVal              time.Duration           `json:"slot_duration_sec"`
+	SlotsPerEpochVal                uint64                  `json:"slots_per_epoch"`
+	EpochsPerSyncCommitteePeriodVal uint64                  `json:"epochs_per_sync_committee_period"`
+}
+
+// NewNetwork creates a new beacon chain network from a parent spec network.
 func NewNetwork(network spectypes.BeaconNetwork) Network {
 	return Network{
-		BeaconNetwork: network,
-		LocalTestNet:  false,
+		Parent: network,
 	}
 }
 
-// NewLocalTestNetwork creates a new local beacon chain network.
-func NewLocalTestNetwork(network spectypes.BeaconNetwork) Network {
-	return Network{
-		BeaconNetwork: network,
-		LocalTestNet:  true,
+func (n Network) String() string {
+	if n.Name != "" {
+		return n.Name
+	}
+
+	if n.Parent != "" {
+		return string(n.Parent)
 	}
+
+	return ""
+}
+
+func (n Network) ForkVersion() [4]byte {
+	if n.ForkVersionVal != ([4]byte{}) {
+		return n.ForkVersionVal
+	}
+
+	if n.Parent != "" {
+		return n.Parent.ForkVersion()
+	}
+
+	return [4]byte{}
 }
 
-// MinGenesisTime returns min genesis time value
 func (n Network) MinGenesisTime() uint64 {
-	if n.LocalTestNet {
-		return 1689072978
+	if n.MinGenesisTimeVal != 0 {
+		return n.MinGenesisTimeVal
+	}
+
+	if n.Parent != "" {
+		return n.Parent.MinGenesisTime()
 	}
-	return n.BeaconNetwork.MinGenesisTime()
+
+	return 0
+}
+
+func (n Network) SlotDurationSec() time.Duration {
+	if n.SlotDurationSecVal != 0 {
+		return n.SlotDurationSecVal
+	}
+
+	if n.Parent != "" {
+		return n.Parent.SlotDurationSec()
+	}
+
+	return 0
```

Comment:
I would return some default insteaf of 0, since this is non functional value, lets take ethereum's default 12. (Put it in a const)